core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -56,21 +65,31 @@ public AtomicFileWriter(File f) throws IOException {
     }
 
     /**
-     * @param encoding
-     *      File encoding to write. If null, platform default encoding is chosen.
+     * @param encoding File encoding to write. If null, platform default encoding is chosen.
+     *
+     * @deprecated Use {@link #AtomicFileWriter(File, Charset)}
      */
     public AtomicFileWriter(File f, String encoding) throws IOException {
-        File dir = f.getParentFile();
+        this(f, Charset.forName(encoding));
```

Comment:
🐛 Violates Javadoc. Will be <code>IllegalArgumentException</code> - https://docs.oracle.com/javase/7/docs/api/java/nio/charset/Charset.html#forName(java.lang.String)

Reply:
@oleg-nenashev we cannot wrap this in a try-catch because it's a constructor call. Do you mean we simply need to enrich the Javadoc, or do we want to wrap this in some way to avoid throwing this kind of exception.

Reply:
you could move the constructor code to an `init()` style method and wrap the CharsetNotFound with an `IOException`

Reply:
@oleg-nenashev @jtnord Addressed: I finally went through a path (no pun intended) that looked better to me. I'm not allowing a null parameter on the new method, and used a ternary on the deprecated shim constructor to defer to the new method.

I'm addressing I think @oleg-nenashev's concern with now a still valid Javadoc: `If null, platform default encoding is chosen.` is still true.
And removed that Javadoc + added `@Nonnull` the new constructor.